### ISSUE TITLE
False positive for ST11 when table name is quoted

### ISSUE THREAD

**Issue by daviewales** [sqlfluff/sqlfluff#6741]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Running `sqlfluf lint` on a file with a quoted table name in a join yields a false positive for ST11.

The same false positive appears when quoting with `[]`.

Replacing `"test-2"` with `test2` prevents the warning from appearing.

### Expected Behaviour

I expect not to get a false positive warning when the table name is quoted.

### Observed Behaviour

I get a false positive warning when the table name is quoted:

```
== [test.sql] FAIL
L:   5 | P:  11 | ST11 | Joined table '"test-2"' not referenced elsewhere in
                       | query [structure.unused_join]
All Finished 📜 🎉!
```

### How to reproduce

Run `sqlfluff lint` on the following file, with the included `.sqlfluff` config:

``` sql
SELECT
    test.one,
    "test-2".two
FROM test
LEFT JOIN "test-2"
    ON test.id = "test-2".id
```

### Dialect

tsql

### Version

sqlfluff, version 3.3.1
Python 3.12.7

### Configuration

``` config
[sqlfluff]
dialect = tsql
max_line_length = 96

# ST06 reorders columns, which seems presumptuous
# CP02 changes capitalisation of fields, which the team agreed we don't want:
exclude_rules = ST06, CP02

[sqlfluff:indentation]
allow_implicit_indents = True

[sqlfluff:rules:references.special_chars]
# Ideally we would not allow special characters in identifiers.
# But we can't control the format of the fields from our CRM
allow_space_in_identifier = True
# I don't want to allow special characters,
# but our CRM is full of fields with these characters.
additional_allowed_characters = ['-/()?,']

# Team decision about capitalisation
[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
